pandas/core/algorithms.py

```diff
@@ -620,6 +622,21 @@ def factorize(
     array([0, 0, 1]...)
     >>> uniques
     Index(['a', 'c'], dtype='object')
+
+    If NaN is in the values, and we want to include NaN in the uniques of the
+    values, it can be achieved by setting ``na_sentinel=None``.
+
+    >>> values = np.array([1, 2, 1, np.nan])
+    >>> codes, uniques = pd.factorize(values)  # default: na_sentinel=-1
+    >>> codes
+    array([ 0,  1,  0, -1])
+    >>> uniques
+    array([1., 2.])
+    >>> codes, uniques = pd.factorize(values, na_sentinel=None)
```

Comment:
ideally put a blank line as othewise this is very hard to read

Reply:
done!